memmrg.h

```diff
@@ -0,0 +1,44 @@
+#ifndef MEMMRG_H
+#define MEMMRG_H
+
+
+typedef int Data;
+
+class MemMrg
+{
+    struct MrgItem
+    {
+        Data *item_val;
+        bool item_used;
+
+        MrgItem():item_used(false)
+        {
+          item_val = new Data;
+        }
+
+        ~MrgItem() {
+            delete item_val;
+        }
+    };
+
+public:
+    MemMrg(const int max_size = 100);
+    ~MemMrg();
+
+    Data* my_alloc(Data init_value = 0);
```

Comment:
void * is usual to use for just memory allocation. It's saying "here is just a raw memory, not an object of any type, use it as you want, ie construct an object there"